functions/check_availability.pp

```diff
@@ -0,0 +1,21 @@
+#
+# @summary check if a group of targets are reachable for bolt
+#
+# @param targets list of targets that are going to be checked
+# @param output_details flag to enable/disable error output for failed nodes
+#
+# @return counter for unavailable nodes
+#
+# @author Tim Meusel <tim@bastelfreak.de>
+#
+function peadm::check_availability(TargetSpec $targets, Boolean $output_details = true) >> Integer {
+  $check_result = wait_until_available($targets, wait_time => 2, _catch_errors => true)
+  if $check_result.ok {
+  } elsif $output_details {
+    $check_result.error_set.each |$result| {
+      out::message("## node ${result.target} has connection error '${result.error.kind}' with message '${result.error.msg}'")
```

Comment:
Maybe you should `fail_plan()` here instead of `out::message()`. That will break the execution. If anyone would like to recover then can run the `peadm::check_availability()` with `_catch_errors => true` option.

Reply:
Then I need to rework it a bit because it's possible that there are multiple unreachable nodes, and I want to catch all of them.